ce_discovery.yaml

```diff
@@ -9,81 +9,230 @@ paths:
          required: true
          schema:
-           type: string
+           $ref: '#/components/schemas/service/properties/id'
+      requestBody:
+        required: false
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/serviceinstance'
       responses:
         '200':
-          description: The corresponding service.
+          description: The Service that was deleted
           content:
             application/json:
               schema:
-                type: array
-                items:
-                  $ref: '#/components/schemas/service'
+                $ref: '#/components/schemas/service'
+        '400':
+          description: Bad Request - constraint failure
+        '404':
+          description: Not Found
+        '409':
+          description: Conflict - epoch not greater
 components:
   schemas:
-    service:
+    eventtype:
+      type: object
+      properties:
+        type:
+          type: string
+          description: CloudEvents type attribute
+          example: com.github.pull.create, com.example.object.delete.v2
+        description:
+          type: string
+          description: Human readable description
+        dataschema:
+          type: string
+          description: CloudEvents datacontenttype attribute. Indicating how the data attribute of subscribed events will be encoded
+        dataschematype:
+          type: string
+          description: If using dataschemacontent for inline schema storage, the dataschematype indicates the type of schema represented there
+        dataschemacontent:
+          type: string
+          description: An inline representation of the schema of the data attribute encoding mechanism. This is an alternative to using the dataschema attribute
+        sourcetemplate:
+          type: string
+          description: A URI Template according to RFC 6570 that defines how the source attribute will be generated
+          example: "http://blob.store/{bucket}"
+        extensions:
+          type: array
+          description: An array or CloudEvents Extension Context Attributes that are used for this event type
+          items:
+            type: object
+            properties:
+              name:
+                type: string
+                description: The CloudEvents context attribute name used by this extension
+              type:
+                type: string
+                description: The data type of the extension attribute
+              specurl:
+                type: string
+                description: An attribute pointing to the specification that defines the extension
+          example: '{ "name": "dataref", "type": "URI-reference", "specurl": "https://github.com/cloudevents/spec/blob/master/extensions/dataref.md" }'
+    eventtypes:
+      type: array
+      items:
+        $ref: '#/components/schemas/eventtype'
+    serviceinstance:
```

Comment:
If I understand this correctly, this is essentially a service _version_? I know it is hard to name these things. The way I usually understand _service instance_ is that there is something like a service broker or a provisioning API that lets you create an instance of a service.

Reply:
You're correct, it [I think] MUST uniquely identify a version of the service record.  `serviceidentifiers` or `serviceversion` may have been a better choice (I'm not attached to this one if you have a preference).  I needed a name with which to reference the schema for a service but only requiring the subset of attributes that uniquely identify the service that was created/updated or is to be deleted.  This is used only here in a the response to `POST /services` and the request body of `DELETE /services`.